test/integration/job/job_test.go

```diff
@@ -3498,14 +3546,40 @@ func cleanUp(ctx context.Context, clientSet clientset.Interface, jobObj *batchv1
 			return err
 		}
 	}
-	return clientSet.BatchV1().Jobs(jobObj.Namespace).Delete(ctx, jobObj.Name, metav1.DeleteOptions{})
+
+	// Set the propagation policy to background to ensure that the job doesn't receive any finalizer at deletion time
+	backgroundDeletion := metav1.DeletePropagationBackground
+	err := clientSet.BatchV1().Jobs(jobObj.Namespace).Delete(ctx, jobObj.Name, metav1.DeleteOptions{
+		PropagationPolicy: &backgroundDeletion,
+	})
+	if err != nil {
+		return err
+	}
+	if sync {
+		err := wait.PollUntilContextTimeout(ctx, waitInterval, 5*time.Second, true, func(ctx context.Context) (bool, error) {
+			_, err := clientSet.BatchV1().Jobs(jobObj.Namespace).Get(ctx, jobObj.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return true, nil
+				} else {
```

Comment:
Else block is reduntant here, it can safely be outdented as the if clause has a return

Reply:
The sync logic might not be needed at all, see this [comment](https://github.com/kubernetes/kubernetes/pull/126439#issuecomment-2272139118).

Reply:
I finally dropped the sync code, see [this comment](https://github.com/kubernetes/kubernetes/pull/126439#issuecomment-2275307459) 😊